### PR TITLE
Display workspace ID in API keys

### PR DESCRIPTION
**Pull request by albandum**
## Description

- People constantly ask us where to find their workspace ID and we always answer "after the w in your URL". This is just bad UX
- Let's make the workspace ID available neatly :) 

Screenshot 2024-09-24 at 16 31 10

### CODE REVIEW

**Comment by PopDaph** [comment]
Discussed IRL will review once Ed's reco is implemented 🙇🏻‍♀️

**Comment by albandum** [comment]
Added modal with workspaceid and API key after creation 

https://github.com/user-attachments/assets/12bb5277-de19-42fb-9c17-6bed61ef09b7

File: front/pages/w/[wId]/vaults/[vaultId]/apps/index.tsx [hunk 1]
```diff
@@ -308,6 +314,49 @@ export function APIKeys({
 
   return (
     <>
+      <Modal
+        isOpen={isNewApiKeyCreatedOpen}
+        title={"API Key Created"}
+        onClose={function (): void {
```
Comment (PopDaph) [inline front/pages/w/[wId]/vaults/[vaultId]/apps/index.tsx:320]: NIT: For consistency, use the syntax more broadly used in the codebase?

```
onClose={() => setIsNewApiKeyCreatedOpen(false)}
```

**Review by PopDaph — APPROVED** [review]
Thanks!
I tried it locally and it works well, I'm a bit worried about the fact if they close the modal there's no way to open it back and it was wondering if maybe we could allow opening it back while we still display the key. 

Approving anyway cause non blocking 🙇🏻‍♀️

**Comment by albandum** [comment]
> Thanks! I tried it locally and it works well, I'm a bit worried about the fact if they close the modal there's no way to open it back and it was wondering if maybe we could allow opening it back while we still display the key.

If the modal is closed, the key is still displayed fully in the list for 10 minutes :)

**Comment by PopDaph** [comment]
> If the modal is closed, the key is still displayed fully in the list for 10 minutes :)

Yes, that is not what I'm saying on my message! I am saying that if you close the modal you cannot reopen it (while you can still access the key) which seems maybe not the best UX to me. But I'm no designer!

**Pull request merged**